### PR TITLE
Commands: Rename based on design meeting decisions

### LINKED ISSUE #2134: NuGet Commands don't work on Windows 7/PowerShell 2.0
I have Win7 x64, VS2013 update 4.
When I install package EntityFramework an error is raised:

PM> install-package EntityFramework.Commands -pre
Attempting to resolve dependency 'EntityFramework.Relational (≥ 7.0.0-beta4)'.
Attempting to resolve dependency 'EntityFramework.Core (≥ 7.0.0-beta4)'.
Attempting to resolve dependency 'Ix-Async (≥ 1.2.3)'.
Attempting to resolve dependency 'Microsoft.Framework.Caching.Memory (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.OptionsModel (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.ConfigurationModel (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.ConfigurationModel.Interfaces (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.DependencyInjection.Interfaces (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.Caching.Interfaces (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.DependencyInjection (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.Logging (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.Framework.Logging.Interfaces (≥ 1.0.0-beta4)'.
Attempting to resolve dependency 'Remotion.Linq (≥ 2.0.0-alpha-002)'.
Attempting to resolve dependency 'System.Collections.Immutable (≥ 1.1.33-beta)'.
Attempting to resolve dependency 'EntityFramework.Relational.Design (≥ 7.0.0-beta4)'.
Attempting to resolve dependency 'EntityFramework.SqlServer.Design (≥ 7.0.0-beta4)'.
Attempting to resolve dependency 'EntityFramework.SqlServer (≥ 7.0.0-beta4)'.
Attempting to resolve dependency 'Microsoft.CodeAnalysis.CSharp (≥ 1.0.0-rc2)'.
Attempting to resolve dependency 'Microsoft.CodeAnalysis.Common (= 1.0.0-rc2)'.
Attempting to resolve dependency 'System.Reflection.Metadata (= 1.0.18-beta)'.
Attempting to resolve dependency 'Microsoft.CodeAnalysis.Analyzers (= 1.0.0-rc2)'.
Installing 'EntityFramework.Commands 7.0.0-beta4'.
You are downloading EntityFramework.Commands from Microsoft, the license agreement to which is available at http://www.microsoft.com/web/webpi/eula/net_library_eula_enu.htm. Check the package for additional dependencies, which may come with their own license agreement(s). Your use of the package and dependencies constitutes your acceptance of their license agreements. If you do not accept the license agreement(s), then delete the relevant components from your device.
Successfully installed 'EntityFramework.Commands 7.0.0-beta4'.
Where-Object : Cannot bind parameter 'FilterScript'. Cannot convert the "Name" value of type "System.String" to type "System.Management.Automation.ScriptBlock".
At C:\!Projects\HG\BioMan-Portrait\packages\EntityFramework.Commands.7.0.0-beta4\tools\init.ps1:3 char:19
+ if (Get-Module | ? <<<<  Name -eq EntityFramework) {
    + CategoryInfo          : InvalidArgument: (:) [Where-Object], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.PowerShell.Commands.WhereObjectCommand
 
You must provide a value expression on the right-hand side of the '-' operator.
At C:\!Projects\HG\BioMan-Portrait\packages\EntityFramework.Commands.7.0.0-beta4\tools\EntityFramework.psm1:80 char:36
+     if ($frameworkName.Identifier - <<<< in '.NETCore', 'WindowsPhoneApp') {
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : ExpectedValueExpression
 
Import-Module : The module to process 'EntityFramework.psm1', listed in field 'ModuleToProcess' of module manifest 'C:\!Projects\HG\BioMan-Portrait\packages\EntityFramework.Commands.7.0.0-beta4\tools\EntityFramework.psd1' was not processed because no valid modul
e was found in any module directory.
At C:\!Projects\HG\BioMan-Portrait\packages\EntityFramework.Commands.7.0.0-beta4\tools\init.ps1:7 char:14
+ Import-Module <<<<  (Join-Path $toolsPath EntityFramework.psd1) -DisableNameChecking
    + CategoryInfo          : ResourceUnavailable: (EntityFramework:String) [Import-Module], FileNotFoundException
    + FullyQualifiedErrorId : Modules_ModuleFileNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand
 
Adding 'EntityFramework.Commands 7.0.0-beta4' to Database.
Successfully added 'EntityFramework.Commands 7.0.0-beta4' to Database.

### LINKED ISSUE #2531: Commands: Review commands syntax, migration method names, etc.
One proposal is below, we should come up with a  proposal and bring it to the design meeting...

This was mentioned in #1972.

We should have two commands: `Apply-Migration` and `Revert-Migration`.

Also, the `Migration.Up()` and `.Down()` methods should be renamed to fully align.

### PR DESCRIPTION
**Pull request by bricelam**
image

See [the design meeting notes](https://github.com/aspnet/EntityFramework/wiki/Design-Meeting-Notes---July-23,-2015#nugetdnx-commands) for more details.

Resolves #2531 & resolves #2134

### CODE REVIEW

File: src/EntityFramework.Commands/Program.cs [hunk 1]
```diff
@@ -64,213 +63,245 @@ public virtual int Main([NotNull] string[] args)
...
-            _app = new CommandLineApplication { Name = "ef" };
-            _app.VersionOption(
-                "-v|--version",
+            var app = new CommandLineApplication
+            {
+                Name = "ef",
+                FullName = "Entity Framework Commands"
+            };
+            app.VersionOption(
+                "--version",
```
Comment (smitpatel) [inline src/EntityFramework.Commands/Program.cs]: -v for version gone?

File: src/EntityFramework.Commands/Program.cs [hunk 2]
```diff
@@ -64,213 +63,245 @@ public virtual int Main([NotNull] string[] args)
...
+                    dbcontext.Description = "Commands to manage your DbContext types";
+                    dbcontext.HelpOption("-?|-h|--help");
+                    dbcontext.OnExecute(() => dbcontext.ShowHelp());
+                    dbcontext.Command(
                         "list",
                         list =>
                         {
-                            list.Description = "List the contexts";
-                            list.HelpOption("-h|--help");
+                            list.Description = "List you DbContxt types";
```
Comment (smitpatel) [inline src/EntityFramework.Commands/Program.cs]: your?

Comment (lajones) [inline src/EntityFramework.Commands/Program.cs]: Typo: you -> your?

Comment (anpete) [inline src/EntityFramework.Commands/Program.cs]: "DbContxt"

File: src/EntityFramework.Commands/Program.cs (hunk 1 above)
Comment (bricelam) [inline src/EntityFramework.Commands/Program.cs]: Yeah, it more commonly means verbose which is to be added later.

File: src/EntityFramework.Commands/Program.cs (hunk 2 above)
Comment (bricelam) [inline src/EntityFramework.Commands/Program.cs]: lol, wow... I fail.

**Comment by lajones** [comment]
:shipit:

**Pull request merged**